src/nvim/statusline.c

```diff
@@ -582,9 +582,10 @@ void win_redr_ruler(win_T *wp)
 
   if (ui_has(kUIMessages) && !part_of_status) {
     MAXSIZE_TEMP_ARRAY(content, 1);
-    MAXSIZE_TEMP_ARRAY(chunk, 2);
+    MAXSIZE_TEMP_ARRAY(chunk, 3);
     ADD_C(chunk, INTEGER_OBJ(attr));
     ADD_C(chunk, CSTR_AS_OBJ(buffer));
+    ADD_C(chunk, INTEGER_OBJ(0));
```

Comment:
Hmm, we probably should either calculate the actual hl_id, or just leave this change out completely (we can add it later if it is actually needed)

Reply:
Right, think I just forgot about this one as a TODO, thanks. In the current logic it is always equal to `HLF_MSG`, added an assert to catch if that ever changes and more logic is needed.